neqo-crypto/src/agent.rs

```diff
@@ -40,16 +43,67 @@ use crate::{
     },
     ech,
     err::{is_blocked, secstatus_to_res, Error, PRErrorCode, Res},
+    experimental_api,
     ext::{ExtensionHandler, ExtensionTracker},
-    null_safe_slice,
     p11::{self, PrivateKey, PublicKey},
     prio,
     replay::AntiReplay,
     secrets::SecretHolder,
-    ssl::{self, PRBool},
+    ssl::{self, PRBool, SSLCertificateCompressionAlgorithm},
     time::{Time, TimeHolder},
 };
 
+experimental_api!(SSL_SetCertificateCompressionAlgorithm(
+    fd: *mut ssl::PRFileDesc,
+    t: SSLCertificateCompressionAlgorithm,
+));
+
+/// The trait is used to represent a certificate compression data structure
+/// Used in order to enable Certificate Compression extension during TLS connection
+pub trait CertificateCompressor {
+    /// Certificate Compression Algorithm identifier
+    /// zlib(1), brotli(2), zstd(3),
+    fn id(&self) -> u16;
+    /// Name of the certificate compression algorithm
+    fn name(&self) -> &'static str;
```

Comment:
```suggestion
    fn name(&self) -> &str;
```

I realize that the natural thing to do is have this be static, but we can have a narrower interface and allow compressors to be configured with strings that they own if they prefer.  That might make testing easier, for example.